src/main/java/repsmax/Deadline.java

```diff
@@ -0,0 +1,44 @@
+package repsmax;
+
+/**
+ * Represents a task with a deadline.
+ * A <code>Deadline</code> object corresponds to a task with a specific due date/time.
+ */
+public class Deadline extends Task {
+
+    /** The date/time by which the task should be completed. */
+    protected String by;
```

Comment:
Any particular reason for using **protected** and not **private**?
```suggestion
    protected String by; Relevance of access modifier
```